### PR TITLE
chore(plugin-swc): bump swc-plugins

### PR DESCRIPTION
**Pull request by JSerFeng**
## Summary

bump swc-plugins. 0.6 minor change is because added react-const-elements

### <samp>🤖 Generated by Copilot at 3d764f9</samp>

This pull request updates the `@modern-js/swc-plugins` dependency and the SWC configuration in the `@modern-js/libuild-plugin-swc` and `@modern-js/builder-plugin-swc` packages. This is to improve the compatibility and performance of the SWC compiler for modern JavaScript projects.

### <samp>🤖 Generated by Copilot at 3d764f9</samp>

*  Update `@modern-js/swc-plugins` dependency to `0.6.0` in both `@modern-js/builder-plugin-swc` and `@modern-js/libuild-plugin-swc` packages ([link](https://github.com/web-infra-dev/modern.js/pull/4502/files?diff=unified&w=0#diff-34ed68e35fc7b09be4c7b86093bb347bb44eb062dc6ad1961198c212ccf433caL67-R67), [link](https://github.com/web-infra-dev/modern.js/pull/4502/files?diff=unified&w=0#diff-9a0096851b44cc1986035406e7425ccac3a66d63726b082014b0b761c2c7a21bL25-R25))
*  Remove unsupported `target: 'es5'` option from default SWC configuration in `@modern-js/builder-plugin-swc` ([link](https://github.com/web-infra-dev/modern.js/pull/4502/files?diff=unified&w=0#diff-0cf91a085824ef67c4d6a4fdbeb226b1620b511f970bf885dfd0cb465e66a863L133))
*  Add changeset file to document the patch updates to both packages in English and Chinese ([link](https://github.com/web-infra-dev/modern.js/pull/4502/files?diff=unified&w=0#diff-f18f53b79875eec8ad46bee187e989e3982c0ec50bb957fe9a59ab8430560f6aR1-R8))

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by chenjiahan — APPROVED** [review]

File: packages/builder/plugin-swc/package.json [hunk 1]
```diff
@@ -64,7 +64,7 @@
     "@babel/preset-react": "^7.18.6",
     "@babel/preset-typescript": "^7.21.5",
     "@modern-js/builder-shared": "workspace:*",
-    "@modern-js/swc-plugins": "0.5.5",
+    "@modern-js/swc-plugins": "0.6.0",
```
Comment (chenjiahan) [inline packages/builder/plugin-swc/package.json:67]: Is `react-const-elements` enabled by default?

Comment (chenjiahan) [reply packages/builder/plugin-swc/package.json:67]: We do not enable it when using babel, and the default behavior of SWC should be same as babel.

Comment (JSerFeng) [reply packages/builder/plugin-swc/package.json:67]: No, it's not enabled by default

Comment (chenjiahan) [reply packages/builder/plugin-swc/package.json:67]: get

**Pull request merged**